### PR TITLE
DBZ-7237 Align LOB behavior in snapshot and streaming

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-7237

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java [hunk 1]
```diff
@@ -324,45 +337,61 @@ protected Object convertBinary(Column column, Field fieldDefn, Object data, Bina
...
-                else {
-                    Blob blob = (Blob) data;
-                    data = blob.getBytes(1, Long.valueOf(blob.length()).intValue());
-                }
-            }
             else if (data instanceof RAW) {
                 data = ((RAW) data).getBytes();
             }
-
-            if (data == UNAVAILABLE_VALUE) {
```
Comment (sherpa003) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java:355]: is this if statement not needed for Binary data type? I don't know, maybe not...

Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java:355]: It was moved to the `convertBlob` method, so it was retained, just moved.

**Comment by jpechane** [comment]
@Naros I really wonder if we should drop unavailable value placeholder. IMHO it is valid content in this context, WDYT?

**Comment by Naros** [comment]
I believe the idea @jpechane was to basically remove any value related to these columns entirely when `lob.enabled` was not set in the configuration; however, I can see the point of maintaining that value as a clear indicator of that being the setting in the connector configs.  

I'd then say we need to tackle this in the reverse and when `lob.enabled` is set to `false` during the snapshot, we should explicitly include the unavailable value placeholder during snapshot to match the behavior of streaming.

@sherpa003 you okay with this, as long as the outcome from streaming/snapshot are aligned?

**Comment by sherpa003** [comment]
> I believe the idea @jpechane was to basically remove any value related to these columns entirely when `lob.enabled` was not set in the configuration; however, I can see the point of maintaining that value as a clear indicator of that being the setting in the connector configs.
> 
> I'd then say we need to tackle this in the reverse and when `lob.enabled` is set to `false` during the snapshot, we should explicitly include the unavailable value placeholder during snapshot to match the behavior of streaming.
> 
> @sherpa003 you okay with this, as long as the outcome from streaming/snapshot are aligned?

Yes, the main goal is to have consistent behavior. So if lob.enabled=false, the placeholder value should be there also for short CLOB (treated as VARCHAR) values and for snapshot values.  Then it should be fine. Thank you

**Comment by Naros** [comment]
Superseded by #5092.

**Pull request closed without merging**